node_version_parser.go

```diff
@@ -0,0 +1,56 @@
+package nodeengine
```

Comment:
Would it actually make sense not to use a separate `NodeVersionParser` at all, and simply repurpose `NvmrcParser` to read from either source, since `NvrmrcParser`'s accepted formats are a superset of `NodeVersionParser`'s?

Reply:
I like `NodeVersionParser` as separate. Given that (1) nvmrc is well defined by the node version manager, but .node-version does not have a format standard and (2) as separate it allows better readability in detect, it makes sense to me.